packages/web3-utils/src/converters.ts

```diff
@@ -86,7 +86,8 @@ export const bytesToBuffer = (data: Bytes): Buffer | never => {
 	}
 
 	if (typeof data === 'string' && isHexStrict(data)) {
-		return Buffer.from(data.slice(2), 'hex');
+		const finalData = data.length % 2 === 0 ? data : `0${data}`;
```

Comment:
This code block executes when `data` is already prefixed with `0x`, so returning `0${data}` will end up wrong hex string. e.g. If `data='0x1` it will return `00x1`.

Also add a few tests for this specific case.

Reply:
I had in mind the input from `numberToHex`, very good point thanks!!
I didn't find any tests for `bytesToBuffer`, I'll add them including this case